examples/threading/threading.c

```diff
@@ -0,0 +1,85 @@
+#include "threading.h"
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+// Optional: use these functions to add debug or error prints to your application
+#define DEBUG_LOG(msg,...)
+//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+#define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
+
+
+void* threadfunc(void* thread_param) {
+
+    // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
+    // hint: use a cast like the one below to obtain thread arguments from your parameter
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    // Convert wait times from milliseconds to seconds
+    int time_before_lock = thread_func_args->delay_before_lock_ms / 1000;
+    int lock_duration = thread_func_args->hold_mutex_duration_ms / 1000;
+
+    // Sleep for the specified time before attempting to lock the mutex
+    sleep(time_before_lock);
```

Comment:
@Buff-Chandan you should handle the return code here and in the following `sleep` call to see if the sleep completed.